activerecord/lib/active_record/core.rb

```diff
@@ -552,6 +552,11 @@ def slice(*methods)
       Hash[methods.flatten.map! { |method| [method, public_send(method)] }].with_indifferent_access
     end
 
+    # Returns an array of the given methods with their values
+    def values_at(*methods)
+      methods.flatten.map! { |method| public_send(method) }
```

Comment:
Does not `flat_map` work here?

Reply:
it does not. Because:

> a.flat_map(&b) works exactly like a.map(&b).flatten!(1).

So if you send an array as parameter, it will try to `public_send` on an array and raise an exception.

I added test for array as params if needed